src/course/views.py

```diff
@@ -134,7 +133,64 @@ def post(self):
                 current_user.courses.append(course)
                 models.db.session.commit()
                 author = models.User.query.filter_by(id=course.teacher_id).first()
-                print course.id
                 return "Redirect to:%s" % (url_for("view_course", courseID=course.id+1000))
         else:
             return "Registration Code Incorrect"
+
+class AddTAView(MethodView):
+
+    def get(self):
+        return
+
+    def post(self):
+        data = flask.request.get_json()
+        email = data.get('email')
+        courseId = data.get('courseID')
+        course = models.Course.query.filter_by(id=courseId).first()
+        if email:
+            user = models.User.query.filter(models.User.email.contains(email)).first()
+            if user and user.permissions:
+                if user.permissions < 20:
+                    user.permissions = 20
+                secondaryTeachers = course.secondaryTeachers
+                userIdAndComma = str(user.id) + ","
+                if secondaryTeachers.find("," + userIdAndComma) >= 0:
+                    return "failure"
+                secondaryTeachers = secondaryTeachers + userIdAndComma
+                course.secondaryTeachers = secondaryTeachers
+                models.db.session.commit()
+                return email
+            else: 
+                return "failure"
+        else:
+            return "failure"
+
+class RemoveTAView(MethodView):
+
+    def get(self):
+        return
+
+    def post(self):
+        data = flask.request.get_json()
+        email = data.get('email')
+        courseId = data.get('courseID')
+        course = models.Course.query.filter_by(id=courseId).first()
+        if email:
+            user = models.User.query.filter(models.User.email.contains(email)).first()
+            if user and user.permissions:
+                tas = models.Course.query.filter(models.Course.secondaryTeachers.contains(","+str(user.id)+",")).all()
```

Comment:
Why are there leading and trailing commas here? are you assuming that we are looking for a teacher in the middle of a list such as author2 in the list "author1,author2,author3"?
If you wanted to look for author1 or author3 this query wouldn't work.